### PR TITLE
Fixes for PR #238

### PR DESCRIPTION
**Pull request by drpatelh**
Refactor code introduced in https://github.com/nf-core/fetchngs/pull/238 to use `collectFile` for merging samplesheets.

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -114,7 +114,7 @@ profiles {
         shifter.enabled        = false
         charliecloud.enabled   = false
         apptainer.enabled      = false
-        runOptions             = '-u $(id -u):$(id -g)'
+        docker.runOptions      = '-u $(id -u):$(id -g) -v $HOME:$HOME'
```
Comment (drpatelh) [inline nextflow.config:117]: Remove this after a fix has been pushed to nf-test upstream

File: nf-test.config [hunk 2]
```diff
@@ -3,7 +3,7 @@ config {
     testsDir "."
 
     // nf-test directory including temporary files for each test
-    workDir "/tmp"
+    workDir "nf-test"
```
Comment (drpatelh) [inline nf-test.config:6]: Revert to `/tmp`. Can't use `.nf-test` due to bug with `fasterq-dump` not being able to output files with `.`

File: tests/nextflow.config [hunk 3]
```diff
@@ -0,0 +1 @@
+docker.runOptions = '-u $(id -u):$(id -g) --platform=linux/amd64 -v $HOME:$HOME'
```
Comment (Midnighter) [inline tests/nextflow.config]: Is mounting HOME required to get this to work? In general I find it to be a huge anti-pattern to do so, as it breaks container encapsulation in my opinion.

Comment (drpatelh) [reply tests/nextflow.config]: Deleted that file and removed the home mounting in https://github.com/nf-core/fetchngs/pull/243/commits/300cb2e6a27fcb5e83a9c09389325ab19d0d3f99 

We added it temporarily to fix the tests due to https://github.com/askimed/nf-test/issues/149

nf-test v0.8.3 seems to work now

**Comment by drpatelh** [comment]
Tests are failing due to https://github.com/askimed/nf-test/issues/173

Think we need to remove using snapshots until that is fixed and add explicit tests for the output files.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -13,7 +13,9 @@ on:
 
 env:
   NXF_ANSI_LOG: false
-  NFTEST_VER: "0.8.1"
+  NFTEST_VER: "0.8.3"
+  NFT_DIFF: "icdiff"
+  NFT_DIFF_ARGS: "-N --cols 200 -L expected -L observed -t"
```
Comment (maxulysse) [inline .github/workflows/ci.yml:18]: What are these new ENV variables?

File: nf-test.config (hunk 2 above)
Comment (maxulysse) [reply nf-test.config:6]: I've been battling with @adamrtalbot on this.
But we finally agreed and came up with:
https://github.com/nf-core/modules/blob/master/.github/workflows/test.yml#L620
https://github.com/nf-core/modules/blob/master/nf-test.config#L6

**Comment by maxulysse** [comment]
Looking nice to me, just think we need to use the env variable for the default location of the nf-test workdir as we do in modules

File: .github/workflows/ci.yml (hunk 4 above)
Comment (drpatelh) [reply .github/workflows/ci.yml:18]: For this reason: https://www.nf-test.com/docs/assertions/snapshots/?h=snaps#example-using-icdiff

**Review by maxulysse — APPROVED** [review]

**Pull request merged**